cilium/cmd/helpers.go

```diff
@@ -385,3 +389,32 @@ func dumpConfig(Opts map[string]string) {
 		}
 	}
 }
+
+// getIpv6EnableStatus api returns the EnableIPv6 status
+// by consulting the cilium-agent otherwise reads from the
+// runtime system config
+func getIpv6EnableStatus() bool {
+	params := daemon.NewGetHealthzParamsWithTimeout(5 * time.Second)
+	brief := true
+	params.SetBrief(&brief)
+	// If cilium-agent is running get the ipv6 enable status
+	if _, err := client.Daemon.GetHealthz(params); err == nil {
+		if resp, err := client.ConfigGet(); err == nil {
+			if resp.Status != nil {
+				return resp.Status.Addressing.IPV6 != nil && resp.Status.Addressing.IPV6.Enabled == true
+			}
+		}
+	} else { // else read the EnableIPv6 status from the file-system
+		agentConfigFile := filepath.Join(defaults.RuntimePath, defaults.StateDir,
+			"agent-runtime-config.json")
+		var agentConfig map[string]interface{}
+
+		if byteValue, err := os.ReadFile(agentConfigFile); err == nil {
+			if err = json.Unmarshal(byteValue, &agentConfig); err == nil {
+				return agentConfig["EnableIPv6"].(bool)
```

Comment:
This is still doing an unchecked type assertion and using a `map` with `string` keys where better safety can be used by defining `agentConfig` to be a `github.com/cilium/cilium/pkg/option.DaemonConfig`.

Reply:
Done!

Reply:
The change isn't what was requested. The file should be unmarshalled into a `DaemonConfig` struct, not add a new field to `DaemonConfig`.

Reply:
Thanks for the input, Have incorporated your comment.